src/web/stats.rs

```diff
@@ -61,68 +90,210 @@ impl Stats
                 {
-                    Ok(t) => 
+                    None => {hit},
+                    Some(s) => 
                     {
-                        let delta = (chrono::offset::Utc::now()-t.to_utc()).num_seconds();
-                        if delta < (stats_config.hit_cooloff_seconds as i64)
+                        let t = DateTime::parse_from_rfc3339(&s);
+                        match t 
                         {
-                            let total_time = start_time.elapsed().as_secs_f64();
-                            let compute_time = compute_start_time.elapsed().as_secs_f64();
+                            Ok(t) => 
+                            {
+                                let delta = (chrono::offset::Utc::now()-t.to_utc()).num_seconds();
+                                if delta < (stats_config.hit_cooloff_seconds as i64)
+                                {
+                                    let total_time = start_time.elapsed().as_secs_f64();
+                                    let compute_time = compute_start_time.elapsed().as_secs_f64();
 
-                            crate::debug(format!
-                            (
-                                "\nTotal stats time:         {} s (Passthrough)\nCompute stats time:       {} s (Passthrough)", 
-                                total_time,
-                                compute_time
-                            ), Some("PERFORMANCE".to_string()));
+                                    crate::debug(format!
+                                    (
+                                        "\nTotal stats time:         {} s (Passthrough)\nCompute stats time:       {} s (Passthrough)", 
+                                        total_time,
+                                        compute_time
+                                    ), Some("PERFORMANCE".to_string()));
 
-                            return
+                                    return
+                                }
+                                hit
+                            },
+                            Err(_e) => {hit}
                         }
-                    },
-                    Err(_e) => {}
+                    }            
+                }
+            },
+            false => 
+            {
+
+                Hit {path: uri, count: 1, times: vec![chrono::offset::Utc::now().to_rfc3339()], ip: ipv4.to_string()}
+            }
+        };
+
+        crate::debug(format!("[Hit] {:?}", hit), None);
+
+        stats.hits.insert(hash, hit);
+
+        let compute_time = compute_start_time.elapsed().as_secs_f64();
+
+        let total_time = start_time.elapsed().as_secs_f64();
+
+        crate::debug(format!
+        (
+            "\nTotal stats time:         {} s\nCompute stats time:       {} s", 
+            total_time,
+            compute_time
+        ), Some("PERFORMANCE".to_string()));
+    }
+
+    fn process_hits(path: String, from: DateTime<chrono::Utc>) -> Digest
```

Comment:
- [ ] use this to process the hits on a save
- [ ] post to Discord every period